action/protocol/staking/ethabi/stake_base.go

```diff
@@ -1,203 +1,25 @@
-			args[i].StakedAmount = amount
-		} else {
-			return "", errConvertBigNumber
-		}
-
-		args[i].AutoStake = bucket.AutoStake
-		addr, err = addrutil.IoAddrToEvmAddr(bucket.Owner)
-		if err != nil {
-			return "", err
-		}
-		args[i].Owner = addr
-
-		if bucket.ContractAddress == "" {
-			// native bucket contract address is 0x0000000000000000000000000000000000000000
-			args[i].ContractAddress = common.Address{}
-			args[i].StakedDuration = bucket.StakedDuration
-			args[i].CreateTime = bucket.CreateTime.Seconds
-			args[i].StakeStartTime = bucket.StakeStartTime.Seconds
-			args[i].UnstakeStartTime = bucket.UnstakeStartTime.Seconds
-		} else {
-			addr, err = addrutil.IoAddrToEvmAddr(bucket.ContractAddress)
-			if err != nil {
-				return "", err
-			}
-			args[i].ContractAddress = addr
-			args[i].StakedDurationBlockNumber = bucket.StakedDurationBlockNumber
-			args[i].CreateBlockHeight = bucket.CreateBlockHeight
-			args[i].StakeStartBlockHeight = bucket.StakeStartBlockHeight
-			args[i].UnstakeStartBlockHeight = bucket.UnstakeStartBlockHeight
-		}
-	}
-
-	data, err := outputs.Pack(args)
-	if err != nil {
-		return "", nil
-	}
-	return hex.EncodeToString(data), nil
-}
-
-func encodeCandidateToEth(candidate *iotextypes.CandidateV2) (*CandidateEth, error) {
-	result := &CandidateEth{}
-	addr, err := addrutil.IoAddrToEvmAddr(candidate.OwnerAddress)
-	if err != nil {
-		return nil, err
-	}
-	result.OwnerAddress = addr
-	addr, err = addrutil.IoAddrToEvmAddr(candidate.OperatorAddress)
-	if err != nil {
-		return nil, err
-	}
-	result.OperatorAddress = addr
-	addr, err = addrutil.IoAddrToEvmAddr(candidate.RewardAddress)
-	if err != nil {
-		return nil, err
-	}
-	result.RewardAddress = addr
-	result.Name = candidate.Name
-	if amount, ok := new(big.Int).SetString(candidate.TotalWeightedVotes, 10); ok {
-		result.TotalWeightedVotes = amount
-	} else {
-		return nil, errConvertBigNumber
-	}
-	result.SelfStakeBucketIdx = candidate.SelfStakeBucketIdx
-	if amount, ok := new(big.Int).SetString(candidate.SelfStakingTokens, 10); ok {
-		result.SelfStakingTokens = amount
-	} else {
-		return nil, errConvertBigNumber
-	}
-	return result, nil
-}
-
-func encodeBucketTypeListToEth(outputs abi.Arguments, bucketTypes *iotextypes.ContractStakingBucketTypeList) (string, error) {
-	args := make([]BucketTypeEth, len(bucketTypes.BucketTypes))
-	for i, bt := range bucketTypes.BucketTypes {
-		args[i] = BucketTypeEth{}
-		if amount, ok := new(big.Int).SetString(bt.StakedAmount, 10); ok {
-			args[i].StakedAmount = amount
-		} else {
-			return "", errConvertBigNumber
-		}
-		args[i].StakedDuration = bt.StakedDuration
-	}
-
-	data, err := outputs.Pack(args)
-	if err != nil {
-		return "", err
-	}
-	return hex.EncodeToString(data), nil
-}
-
 // BuildReadStateRequest decode eth_call data to StateContext
 func BuildReadStateRequest(data []byte) (protocol.StateContext, error) {
 	if len(data) < 4 {
-		return nil, errInvalidCallData
+		return nil, stakingComm.ErrInvalidCallData
```

Comment:
nit:
this is checked in v1 and v2's func, no need to check here?

Reply:
okay